packages/ui/ImageItem/MemoPropsDecorator.js

```diff
@@ -0,0 +1,97 @@
+import hoc from '@enact/core/hoc';
+import pick from 'ramda/src/pick';
+import omit from 'ramda/src/omit';
+import React from 'react';
+
+const MemoPropsThemeContext = React.createContext();
+const MemoPropsContext = React.createContext();
+
+/**
+ * Default config for `MemoPropsDecorator`.
+ *
+ * @memberof ui/ImageItem.MemoPropsDecorator
+ * @hocconfig
+ */
+const defaultConfig = {
+	/**
+	 * Array of the props passed throught a context.
+	 *
+	 * @type {String[]}
+	 * @default	[]
+	 * @private
+	 * @memberof  ui/ImageItem.MemoPropsDecorator.defaultConfig
+	 */
+	filter: []
+};
+
+/**
+ * A higher-order component that is used to provide a context provider.
+ * The props which keys are defined in the array of the `filter` config,
+ * will be passed through a context. The other props will be passed
+ * as props.
+ *
+ * @class MemoPropsDecorator
+ * @memberof ui/ImageItem
+ * @hoc
+ * @public
+ */
+const MemoPropsDecorator = hoc(defaultConfig, (config, Wrapped) => {
+	const {filter} = config;
+
+	// eslint-disable-next-line no-shadow
+	function MemoPropsDecorator (props) {
+		const picked = pick(filter, props) || {};
+		const omitted = omit(filter, props) || {};
+
+		return (
+			<MemoPropsContext.Provider value={picked}>
```

Comment:
This will force the component to update every time it receives props, even if the values are exactly the same, because the object has changed and context will force the update.  Some form of memoization should be done if this truly is the desired solution.